core/src/include/Xios.hpp

```diff
@@ -28,34 +28,91 @@ class Xios : public Configured<Xios> {
     Xios();
     ~Xios();
 
-    void finalize();
+    void close_context_definition();
     void context_finalize();
+    void finalize();
     bool isInitialized();
 
     void configure() override;
     void configureServer();
-    void configureCalendar();
+    void configureCalendar(); // TODO?
+
+    /* Date and duration */
+    std::string convertXiosDatetimeToString(cxios_date datetime, bool isoFormat = true);
+    void printCXiosDate(cxios_date date);
+    void printCXiosDuration(cxios_duration duration);
 
+    /* Calendar */
     cxios_date getCalendarOrigin();
     cxios_date getCalendarStart();
-    xios::CDate getCurrentDate();
     cxios_duration getCalendarTimestep();
+    void getCalendarConfiguration(); // TODO?
+    int getCalendarStep();
+    std::string getCurrentDate(bool isoFormat = true);
     void setCalendarOrigin(cxios_date origin);
     void setCalendarStart(cxios_date start);
     void setCalendarTimestep(cxios_duration timestep);
-
-    void getCalendarConfiguration();
-
-    std::string getCalendarDate(bool isoFormat = true);
-    int getCalendarStep();
-
     void updateCalendar(int stepNumber);
-    void write(const std::string fieldstr, double* data, const int ni, const int nj);
-
-    std::string convertXiosDatetimeToString(cxios_date datetime, bool isoFormat = true);
 
-    void printCXiosDate(cxios_date date);
-    void printCXiosDuration(cxios_duration duration);
+    /* Axis */
+    void createAxis(std::string axisId); // TODO
+    void setAxisSize(std::string axisId, int size); // TODO
+    void setAxisValues(std::string axisId, std::vector<double> values); // TODO
+    int getAxisSize(std::string axisId);
+    std::vector<double> getAxisValues(std::string axisId);
+
+    /* Domain */
+    void createDomain(std::string domainId); // TODO
+    void setDomainType(std::string domainId, std::string domainType); // TODO
+    void setDomainGlobalLongitudeSize(std::string domainId, int size); // TODO
+    void setDomainGlobalLatitudeSize(std::string domainId, int size); // TODO
+    void setDomainLongitudeSize(std::string domainId, int size);
+    void setDomainLatitudeSize(std::string domainId, int size);
+    void setDomainLongitudeStart(std::string domainId, int start);
+    void setDomainLatitudeStart(std::string domainId, int start);
+    void setDomainLongitudeValues(std::string domainId, std::vector<double> values);
+    void setDomainLatitudeValues(std::string domainId, std::vector<double> values);
```

Comment:
I did end up implementing a few setters (here) because they are quantities that are local to the MPI process.

Reply:
Thanks. Yeah I figured I've probably wrapped some parts of the API we won't end up using but I thought it was better to do it this way and prune it later, rather than having to do more wrapping.

Reply:
Good point on using neXtSIM-DG's data types (although I thought it would be `ModelArray` rather than `ModelArrayRef`). Perhaps if we merge this PR first then #545 could be updated to do this for these setters as well as the `write` method(s).

Reply:
sounds like a good idea